Source/Libraries/openXDA.Nodes/Types/Analysis/AnalysisNode.cs

```diff
@@ -306,19 +312,51 @@ private void Process(MeterDataSet meterDataSet)
                     }
 
                     Log.Debug($"Finished executing data operation '{dataOperationDefinition.UnqualifiedTypeName}'.");
+                    if (index == 0) status = FileGroupProcessingStatus.Success;
+                    else if (status == FileGroupProcessingStatus.Failed) status = FileGroupProcessingStatus.PartialSuccess;
                 }
                 catch (Exception ex)
                 {
                     // Log the error and skip to the next data operation
                     string message = $"An error occurred while executing data operation of type '{dataOperationDefinition.TypeName}' on data from meter '{meterDataSet.Meter.AssetKey}': {ex.Message}";
                     Exception wrapper = new Exception(message, ex);
+                    if (status == FileGroupProcessingStatus.Success) status = FileGroupProcessingStatus.PartialSuccess;
                     Log.Error(wrapper.Message, wrapper);
+                    dataOperationFailures.Add(new DataOperationFailure
+                    {
+                        DataOperationID = dataOperationDefinition.ID,
+                        FileGroupID = meterDataSet.FileGroup.ID,
+                        Log = message,
+                        StackTrace = ex.StackTrace,
+                        TimeOfFailure = DateTime.UtcNow
+                    });
+                    if (IsDisposed)
```

Comment:
If we dispose we need to consider the entire Processing `Failed`.
`PartialSuccess` should only apply to failures inside 
```
 meterDataSet.Configure(dataOperation);
 dataOperation.Execute(meterDataSet);
 ```
 If the underlying infrastiurcture `AnalysisNode` fails that is considered a complete failure since It will not even attepmt to run the rest of the analytics

Reply:
This makes sense, updated.